designs/in-place-upgrades-eksa.md

```diff
@@ -98,15 +149,15 @@ The node upgrade process we need to perform, although different depending on the
 2. Upgrade containerd.
 3. Upgrade CNI plugins.
 4. Update kubeadm binary and run the `kubeadm upgrade` process.
-5. Drain the node.
+5. (Optional) Drain the node.
 6. Update `kubectl`/`kubelet` binaries and restart the kubelet service.
-7. Uncordon de node.
+7. Uncordon the node.
 
 Each of this steps will be executed as an init container in a privileged pod. For the commands that need to run "on the host", we will use `nsenter` to execute them in the host namespace. All these steps will be idempotent, so the full pod can be recreated and execute all the containers from the start or only a subset of them if required.
 
 Draining and uncordoning the node could run in either the container or the host namespace. However, we will run it in the container namespace to be able to leverage the injected credentials for the `ServiceAccount`. This way we don't depend on having a kubeconfig in the host disk. This not only allows us to easily limit the rbac permissions that the `kubectl` command will use, but it's specially useful for worker nodes, since these don't have a kubeconfig with enough permissions to perform these actions (CP nodes have an admin kubeconfig).
 
-In order to codify the logic of each step (the ones that require logic, like the kubeadm upgrade), we will build a single go binary with multiple commands (one per step). The `ControlPlaneKubeadmUpgrade` and `WorkersKubeadmUpgrade` will just reference these commands when building the init containers spec.
+In order to codify the logic of each step (the ones that require logic, like the kubeadm upgrade), we will build a single go binary with multiple commands (one per step). The `ControlPlaneUpgrade` and `MachineDeploymentUpgrade` will just reference these commands when building the init containers spec.
```

Comment:
update: add a line about script temporarily instead of binary.
Update the last sentence to say `NodeUpgrade` instead of CPUpgrade and MDUpgrade